lib/model/folderstate.go

```diff
@@ -24,6 +24,7 @@ const (
 	FolderSyncing
 	FolderCleaning
 	FolderCleanWaiting
+	FolderOutOfSpace
 	FolderError
```

Comment:
There's no need for a new statue: Being out of space is already part of `FolderError`. And the concrete error that caused entering that state is already accessible (part of the event and returned by `.getState`). One of the reasons can be an out-of-space error, among others.